### ISSUE TITLE
[3.3] Conflict with symfony/framework-bundle 6.4.6 || 7.0.6

### ISSUE THREAD

**Issue by lermontex** [api-platform/core#6310]
**API Platform version(s) affected**: v3.3.0-beta.2

**Description**  
I'm using Symfony 7.0.6 and api-platform/core v3.3.0-beta.1. When trying to update to version v3.3.0-beta.2 an error occurs:

```
Loading composer repositories with package information
Updating dependencies                                 
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires api-platform/core v3.3.0-beta.2 -> satisfiable by api-platform/core[v3.3.0-beta.2].
    - api-platform/core v3.3.0-beta.2 conflicts with symfony/framework-bundle 6.4.6 || 7.0.6.
    - symfony/framework-bundle is locked to version v7.0.6 and an update of this package was not requested.
```

I understand that this is related to PR #6292, however, please note that the change in Symfony 7.0.6 does not cause an error when working with GraphQL. My understanding is that this applies to using Enum as an API resource.

**How to reproduce**  
Use Symfony 7.0.6 and install the api-platform/core v3.3.0-beta.2 package

**Possible Solution**  
Allow the use of api-platform/core v3.3.0-beta.2 along with Symfony 6.4.6/7.0.6

**Additional Context**  

https://github.com/api-platform/core/blob/1a120ee98db27b9fb3237637579c52e80fe6fadd/composer.json#L111

**Comment by soyuka** [comment]
can you stay on 7.0.5 until 7.0.7 is released?

**Comment by lermontex** [comment]
I'm currently using Symfony 7.0.6 and api-platform/core v3.3.0-beta.1 and haven't noticed any breaking bugs when working with GraphQL (including Enum).

Since the update was made for testing, I can continue to use previous versions. I can wait for the next minor version of Symfony.

Thank you!

**Comment by michaljusiega** [comment]
See https://github.com/symfony/symfony/pull/54148 - probably we talk about this PR.

**Comment by lermontex** [comment]
The issue was solved sience release of symfony 7.0.7 + api-platform/core 3.0.0
